### ISSUE TITLE
Fetching cloud-config and initial-cluster from S3 fails in newer regions e.g. Frankfurt

### ISSUE THREAD

**Issue by wombat** [xuwang/aws-terraform#25]
I tried using this project in the Frankfurt datacenter but it fails while fetching the cloud-config and initial-cluster because this datacenter uses the newer AWS4 authorization:

```
cat ${workDir}/cloud-config.yaml 
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>InvalidRequest</Code><Message>The authorization mechanism you have provided is not supported. Please use AWS4-HMAC-SHA256.</Message><RequestId>26B794E2DF6DBB38</RequestId><HostId>1MUzjCurL7ZI8zEi5D/fNUUrSdRo7WMJnrH/I+ujSuPogkuHIOJztpzXWQQ9diOlwE5xQYb1Rrw=</HostId></Error>
```

**Comment by xuwang** [comment]
@w0mbat 

Try upgrade your awscli to the latest version. If it still doesn't help, do
this:
```
aws configure set default.s3.signature_version s3v4
```
And also make sure in envs.sh, the default region is set to your region:

```
export AWS_PROFILE=my-aws-profile
export CLUSTER_NAME=my-cluster
export AWS_REGION=eu-central-1
```

Let me know if this helps.

**Comment by w0mbat** [comment]
@xu

**Comment by wombat** [comment]
@xuwang I dont think this will help.

I already have the latest cli and the error happens on one of the etcd machines on aws and not on my local machine.

I am talking about this script: https://github.com/xuwang/aws-terraform/blob/master/resources/cloud-config/s3-cloudconfig-bootstrap.sh

It fires 2 curl commands that use the older v2 authorization

**Comment by xuwang** [comment]
@w0mbat ha I missed that. The script needs to be updated to support sig-v4. I will work on it and get it test. Thanks!

**Comment by xuwang** [comment]
@w0mbat Committed https://github.com/xuwang/aws-terraform/commit/5fa207000625751920415bceef98bc9d3dd4a45a with new s3-cloudconfig-bootstrap.sh to support sig-v4. We have tested it with this configuration in envs.sh:
```
...
export AWS_REGION=eu-central-1
export COREOS_UPDATE_CHANNEL=beta
...
```
It seems working for bootstrapping the cluster. The cloud config file downloaded are in /root/cloudinit directory, as well as logs. 

Note that we have to use CoreOS beta image. The latest (just released today) with docker 1.12.6 seems having container networking issue that we are trying to reproduce and report the problem.

**Comment by xuwang** [comment]
fixed by https://github.com/xuwang/aws-terraform/commit/5fa207000625751920415bceef98bc9d3dd4a45a
